bridge-validators/src/block.rs

```diff
@@ -54,38 +136,56 @@ impl BlockPolling for ChainClient {
         Ok(())
     }
 
-    async fn get_events<D>(
-        &self,
-        event: Filter,
-        from_block: BlockNumber,
-        to_block: BlockNumber,
-    ) -> Result<Vec<D>>
+    async fn get_events<D>(&self, event: Filter, from_block: u64, to_block: u64) -> Result<Vec<D>>
     where
         D: EthEvent,
     {
-        let event = event.from_block(from_block).to_block(to_block);
-
-        let logs: Vec<serde_json::Value> = self
-            .client
-            .provider()
-            .request("eth_getLogs", [event])
-            .await?;
+        let event = event
+            .from_block(from_block)
+            .to_block(to_block)
+            .address(self.chain_gateway_address);
 
-        let logs = logs
-            .into_iter()
-            .map(|log| {
-                // Parse log values
-                let mut log = log;
-                match log["removed"].as_str() {
-                    Some("true") => log["removed"] = serde_json::Value::Bool(true),
-                    Some("false") => log["removed"] = serde_json::Value::Bool(false),
-                    Some(&_) => warn!("invalid parsing"),
-                    None => (),
-                };
-                let log: Log = serde_json::from_value(log)?;
-                Ok(log)
-            })
-            .collect::<Result<Vec<Log>>>()?;
+        let logs: Vec<Log> = match self.log_strategy {
+            LogStrategy::GetLogs => {
+                let logs: Vec<serde_json::Value> = self
+                    .client
+                    .provider()
+                    .request("eth_getLogs", [event])
+                    .await?;
+                logs.into_iter()
+                    .filter(|log| {
+                        log.get("address")
+                            .and_then(|val| val.as_str())
+                            .and_then(|val| val.parse::<Address>().ok())
+                            .map(|from_address| {
+                                if from_address == self.chain_gateway_address {
+                                    true
+                                } else {
+                                    info!(
+                                        "event from {0:#x} , chain gateway {1:#x}",
+                                        from_address, self.chain_gateway_address
+                                    );
+                                    false
```

Comment:
As a double-check against issues like the previous one, where we thought we'd asked ethers to filter logs by source address, but in fact hadn't. The level of impact suggests that we should be careful, and since I'd already implemented this check I didn't feel it was worth removing.